core/src/test/java/com/linecorp/armeria/client/endpoint/StickyStrategyTest.java

```diff
@@ -0,0 +1,109 @@
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.client.endpoint.StickyStrategy.RequestContextHasher;
+import com.linecorp.armeria.common.DefaultHttpRequest;
+import com.linecorp.armeria.common.HttpRequest;
+
+import io.netty.util.AsciiString;
+
+public class StickyStrategyTest {
+
+    private static final String STICKY_HEADER_NAME = "USER_COOKIE";
+
+    RequestContextHasher hasher = (ClientRequestContext ctx) -> {
+            return ((HttpRequest) ctx.request()).headers().get(AsciiString.of(STICKY_HEADER_NAME)).hashCode();
+        };
+    final StickyStrategy strategy = new StickyStrategy(hasher);
+
+    private static final EndpointGroup ENDPOINT_GROUP = new StaticEndpointGroup(
+            Endpoint.of("localhost:1234"),
+            Endpoint.of("localhost:2345"),
+            Endpoint.of("localhost:3333"),
+            Endpoint.of("localhost:5555"),
+            Endpoint.of("localhost:3444"),
+            Endpoint.of("localhost:9999"),
+            Endpoint.of("localhost:1111")
+    );
+
+    private static final EndpointGroup EMPTY_ENDPOINT_GROUP = new StaticEndpointGroup();
+
+    @Before
+    public void setup() {
+        EndpointGroupRegistry.register("endpoint", ENDPOINT_GROUP, strategy);
+        EndpointGroupRegistry.register("empty", EMPTY_ENDPOINT_GROUP, strategy);
+    }
+
+    @Test
+    public void select() {
+        final EndpointSelector selector = strategy.newSelector(ENDPOINT_GROUP);
+        final int selectTime = 5;
+
+        Endpoint ep1 = EndpointGroupRegistry.selectNode(
+                contextWithHeader(STICKY_HEADER_NAME, "armeria1"), "endpoint");
+        Endpoint ep2 = EndpointGroupRegistry.selectNode(
+                contextWithHeader(STICKY_HEADER_NAME, "armeria2"), "endpoint");
+        Endpoint ep3 = EndpointGroupRegistry.selectNode(
+                contextWithHeader(STICKY_HEADER_NAME, "armeria3"), "endpoint");
+
+        // select few times to confirm that same header will be routed to same endpoint
+        for (int i = 0; i < selectTime; i++) {
+            assertThat(EndpointGroupRegistry.selectNode(
+                contextWithHeader(STICKY_HEADER_NAME, "armeria1"), "endpoint")).isEqualTo(ep1);
+            assertThat(EndpointGroupRegistry.selectNode(
+                contextWithHeader(STICKY_HEADER_NAME, "armeria2"), "endpoint")).isEqualTo(ep2);
+            assertThat(EndpointGroupRegistry.selectNode(
+                contextWithHeader(STICKY_HEADER_NAME, "armeria3"), "endpoint")).isEqualTo(ep3);
+        }
+
+        //confirm rebuild tree of dynamic
+        DynamicEndpointGroup dynamicEndpointGroup = new DynamicEndpointGroup();
+        EndpointGroupRegistry.register("dynamic", dynamicEndpointGroup, strategy);
+        Endpoint ep4 = Endpoint.of("localhost:1111");
+        dynamicEndpointGroup.addEndpoint(ep4);
+
+        assertThat(EndpointGroupRegistry.selectNode(
+                contextWithHeader(STICKY_HEADER_NAME, "armeria1"), "endpoint")).isEqualTo(ep4);
```

Comment:
`endpoint` group -> `dynamic` group? Because `ep4` was added to `dynamicEndpointGroup` which was registered as `dynamic` group.